judgels-backends/judgels-commons/judgels-service-core/src/main/java/judgels/service/ServiceUtils.java

```diff
@@ -60,6 +64,38 @@ public static Response buildDownloadResponse(String fileUrl) {
         }
     }
 
+    public static Response buildImageResponseFromText(String text, Date lastModifiedStream) {
+        int fontSize = 14;
+        int margin = 30;
+        int charWidth = 8;
+        int charHeight = 17;
+
+        String[] textList = text.split("\n");
+        Font font = new Font(Font.MONOSPACED, Font.PLAIN, fontSize);
+        int longestText = Arrays.asList(textList).stream().map(String::length).max(Integer::compareTo).get();
+        int width = charWidth * longestText + 2 * margin;
+        int height = charHeight * textList.length + 2 * margin;
+
+        BufferedImage img = new BufferedImage(width, height, BufferedImage.TYPE_INT_ARGB);
+        Graphics2D g2d = img.createGraphics();
+        g2d.setFont(font);
```

Comment:
Need to set antialiasing for better results. See https://stackoverflow.com/q/28591204